lib/driver.js

```diff
@@ -352,26 +354,53 @@ class YouiEngineDriver extends BaseDriver {
         {logger.info('Connecting to WebDriver: ' + HOST + ':' + PORT);}
 
         this.socket = new net.Socket();
+        this.socket.setTimeout(5000);
+        this.socket.setKeepAlive(true, 1000);
+
+        let socketClient = this.socket;
+
+        let removeListenerHandler = function () {
+          logger.info('Removing listeners');
+          socketClient.removeListener('timeout', timeoutHandler);
+          socketClient.removeListener('close', closeHandler);
+          socketClient.removeListener('end', endHandler);
+          socketClient.removeListener('error', errorHandler);
+        };
 
         // Add an 'error' event handler for the client socket
-        this.socket.on ('error', function (ex) {
+        let errorHandler = function (ex) {
           logger.error(ex);
           logger.error('Check that WebDriver is enabled in application, if a device ensure the proper IP address is used.');
+          removeListenerHandler();
+          socketClient.destroy(); //TODO: should we use or add socketClient.end();
```

Comment:
no need to call `end()`, because the close event is not emitted until all connections are ended.
see: https://nodejs.org/api/net.html#net_event_close